server/fishtest/schemas.py

```diff
@@ -56,18 +57,27 @@
 pgn_file = glob("*.pgn", name="pgn_file")
 even = div(2, name="even")
 datetime_utc = intersect(datetime, fields({"tzinfo": timezone.utc}))
+zlib_data = intersect(Binary, magic("application/zlib", name="zlib_data"))
```

Comment:
I wonder if `application/gzip` / `gzip_data` are more appropriate than `application/zlib` / `zlib_data`
We switched from zlib to gzip with https://github.com/official-stockfish/fishtest/pull/1823

Reply:
https://datatracker.ietf.org/doc/html/rfc6713 says application/gzip is a file format and application/zlib is a stream format. They use the same compression algorithm. Very confusing…

Reply:
gzip, zlib, zip are all using the same compressing algorithm. The problem with zlib is that you need some code to decompress the binary blob.
gzip has some nice propreties:
1. can be cat `cat file0.gz file1.gz > files.gz`. I'm using this to stream all the run's PGNs.gz in only one PGN.gz
2. the browser decompress a gzipped file on the fly (nginx can be configured to decompress for a client unable to decompress)

Reply:
Yes but the data in pgnsdb is application/zlib. I cannot say that it is application/gzip when it isn’t…

Reply:
I wonder if that is due to the `Binary(pgn_zip)`

Reply:
I am going to double check.

Reply:
I double checked. Here is the code:
```
import magic
import pymongo

if __name__ == "__main__":
    client = pymongo.MongoClient()
    pgnsdb = client["fishtest_new"]["pgns"]
    pgns = pgnsdb.find({})
    for pgn in pgns:
        zip = pgn["pgn_zip"]
        mime = magic.from_buffer(zip, mime=True)
        magic_ = magic.from_buffer(zip)
        print(mime, magic_)
```

Reply:
https://github.com/official-stockfish/fishtest/blob/e6254159efefc88824fdee19e0f4508afd422626/worker/worker.py#L1456-L1481

Reply:
I looked at the `pgn_zip` data and the first byte is `0x78` which according to https://datatracker.ietf.org/doc/html/rfc6713 is part of the magic for `application/zlib`. `application/gzip` starts with `0x1f`. Mysterious.

Reply:
`gzip` python module bugs? I was doing this to recover the datetime because the function was not implemented.
```python
                        # Extract and convert the 4 bytes starting at index 4
                        tarinfo.mtime = int.from_bytes(pgn_zip[4:8], byteorder="little")
                        tarf.addfile(tarinfo, io.BytesIO(pgn_zip))
```

Reply:
I now saved the "pgn_zip" data to files and and inspected it with the "file" utility. It still says `application/zlib`. I think my pgn collection contains recent pgns, but just to be sure, perhaps you could also look at it on DEV or PROD?

Reply:
On DEV is `application/gzip` running your script:
```
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-623.pgn", last modified: Sat Apr 27 19:37:23 2024, max compression
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-624.pgn", last modified: Sat Apr 27 19:48:22 2024, max compression
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-625.pgn", last modified: Sat Apr 27 20:53:44 2024, max compression
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-626.pgn", last modified: Sat Apr 27 21:05:36 2024, max compression
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-627.pgn", last modified: Sat Apr 27 21:20:15 2024, max compression
application/gzip gzip compressed data, was "662a442b6c6057d3bbd2978f-628.pgn", last modified: Sat Apr 27 21:25:53 2024, max compression

```